tests/unit/test_sqlite_driver.py

```
@@ -237,3 +239,29 @@ def test_hooks_structure(self):
+	the one add_result path — no persist_execution_items shim."""
+
+	def test_on_item_registered_for_all_runner_types(self):
+		from secator.hooks import sqlite as mod
+		from secator.runners import Scan, Task, Workflow
+		for cls in (Task, Workflow, Scan):
+			self.assertIn('on_item', mod.HOOKS[cls])
+			self.assertIn(mod.update_finding, mod.HOOKS[cls]['on_item'])
+
+	def test_workflow_execution_item_persisted_via_on_item(self):
+		from secator.hooks import sqlite as mod
+		from secator.output_types import Warning
```

Ruff (0.15.21)

[error] Import `Warning` is shadowing a Python builtin

(A004)